File: y2024_C#/AdventOfCode/Day_02/DayTwo.cs
using AdventofCode._General;

namespace AdventOfCode.Day_02;

public class DayTwo : Day {
    private readonly List<Report> _reports;

    public DayTwo() : base("input_02.txt") {
        _reports = [];
        Parse();
    }

    private void Parse() {
        foreach (var levels in Input
                     .Select(line => line.Split(" "))
                     .Select(integerStrings => integerStrings
                         .Select(int.Parse)
                         .ToList())
                ) {
            var report = new Report(levels);
            _reports.Add(report);
        }
    }

    public override int PartOne() {
        CountSafeReports();

        return ResultPartOne;
    }

    private void CountSafeReports() {
        var safeReports = _reports.Count(report => report.IsSafe());
        ResultPartOne = safeReports;
    }

    public override int PartTwo() {
        CountSafeWithProblemDampenerReports();

        return ResultPartTwo;
    }

    private void CountSafeWithProblemDampenerReports() {
        var safeReports = _reports.Count(report => report.IsSafeWithProblemDampener());
        ResultPartTwo = safeReports;
    }
}

File: y2025_Java/src/day02/DayTwo.java
package day02;

import _general.Day;

public class DayTwo extends Day {
    private final InvalidIdFinder invalidIdFinder;

    public DayTwo() {
        this.invalidIdFinder = new InvalidIdFinder(input);
    }

    public static void main(String[] args) {
        var dayTwo = new DayTwo();
        dayTwo.printResults();
    }

    @Override
    protected String partOne() {
        return invalidIdFinder.sumUpInvalidIds();
    }

    @Override
    protected String partTwo() {
        return invalidIdFinder.sumUpInvalidStrictIds(); // 66500951798
    }
}
